### PR TITLE
feat: Game of Life

### PR DESCRIPTION
**Pull request by khod-naderi**
#### References
Add Game of Life to game

#### Checklist

- [x] Added description of change
- [x] Added file name matches [game_of_life](https://github.com/khod-naderi/C/blob/master/games/game_of_life.c)
- [x] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by ayaankhan98 — COMMENTED** [review]
please update the documentation according to doxygen documentation guidelines.

**Comment by khod-naderi** [comment]
Excuse me, my English servant is not good, if you can say it easier

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Code is not up to the [repository standards](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md).
Please read them **carefully** and **follow them**.

**Comment by khod-naderi** [comment]
I added a description to the code, is there still a problem?

File: games/game_of_life.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    
+    // return new generation
+    return newWorld;
+}
+
+
+
+/**
+ * Main program function.
+ * @returns 0 on clean exit.
```
Comment (Panquesito7) [inline games/game_of_life.c:103]:
```suggestion
 * @brief Main function.
 * @returns 0 on exit
```

File: games/game_of_life.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        for(int x=0; x<MAX_X; x++){
+            printf("%c", (*(world + MAX_X*y + x) == 1) ? '@' : ' ' ); 
+        }
+        printf("\n");
+    } 
+}
+
+
+/**
+ * em program function.
```
Comment (Panquesito7) [inline games/game_of_life.c:45]: Please add a proper function description.

File: games/game_of_life.c [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ * @returns char pointer new generation on clean exit.
```
Comment (Panquesito7) [inline games/game_of_life.c:46]: Missing `world` parameter documentation.

File: games/game_of_life.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+#include <stdlib.h>
+#include <time.h>
+
+// define Make fixed values
+#define MAX_X 60 // End point X
+#define MAX_Y 20 // End point Y
+
+
+/**
+ * draw program function. (void)
```
Comment (Panquesito7) [inline games/game_of_life.c:22]: Please add a proper function description.

Comment (Panquesito7) [inline games/game_of_life.c:22]: Missing `world` parameter documentation.

File: games/game_of_life.c [hunk 5]
```diff
@@ -0,0 +1,139 @@
+/**
+* Conway's Game of Life
+* @file game_of_life.c
```
Comment (Panquesito7) [inline games/game_of_life.c:3]:
```suggestion
 * @file
```

File: games/game_of_life.c [hunk 6]
```diff
@@ -0,0 +1,139 @@
+/**
+* Conway's Game of Life
```
Comment (Panquesito7) [inline games/game_of_life.c:2]:
```suggestion
 * @brief [Game of life](https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life)
 * implementation
 *
 * @details The Game of Life, also known simply as Life, is a cellular automaton devised by the British
 * mathematician John Horton Conway in 1970. It is a zero-player game, meaning that its evolution is determined
 * by its initial state, requiring no further input. One interacts with the Game of Life by creating an initial
 * configuration and observing how it evolves. It is Turing complete and can simulate a universal constructor or any other Turing machine.
```

File: games/game_of_life.c [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+* @file game_of_life.c
+* @author [ali naderi](github.com/khod-naderi)
+* @author [Conway](https://en.wikipedia.org/wiki/John_Horton_Conway)
```
Comment (Panquesito7) [inline games/game_of_life.c:5]: The `@author` is for those who made the file (you).
```suggestion
```

File: games/game_of_life.c [hunk 8]
```diff
@@ -0,0 +1,139 @@
+/**
+* Conway's Game of Life
+* @file game_of_life.c
+* @author [ali naderi](github.com/khod-naderi)
```
Comment (Panquesito7) [inline games/game_of_life.c:4]:
```suggestion
 * @author [Ali Naderi](https://github.com/khod-naderi)
```

File: games/game_of_life.c [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+* @author [Conway](https://en.wikipedia.org/wiki/John_Horton_Conway)
+* @brief [game of life game](https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life)
+* implementation in C
+* @details  game of life, zero player
+* It has a general array called world, which is the game world in which cells live
```
Comment (Panquesito7) [inline games/game_of_life.c:9]:
```suggestion
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Good code.

File: games/game_of_life.c [hunk 10]
```diff
@@ -0,0 +1,146 @@
+/**
+ * @brief [Game of life](https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life)
+ * implementation
+ *
+ * @details The Game of Life, also known simply as Life, is a cellular automaton devised by the British
+ * mathematician John Horton Conway in 1970. It is a zero-player game, meaning that its evolution is determined
+ * by its initial state, requiring no further input. One interacts with the Game of Life by creating an initial
+ * configuration and observing how it evolves. It is Turing complete and can simulate a universal constructor or any other Turing machine.
+ * @file
```
Comment (Panquesito7) [inline games/game_of_life.c:9]:
```suggestion
```

File: games/game_of_life.c [hunk 11]
```diff
@@ -0,0 +1,146 @@
+/**
```
Comment (Panquesito7) [inline games/game_of_life.c:1]:
```suggestion
/**
 * @file
```

**Comment by khod-naderi** [comment]
I fix warning on Ubuntu, Mac and Windows

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please address **all** the unresolved conversations above.

File: games/game_of_life.c [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+ * mathematician John Horton Conway in 1970. It is a zero-player game, meaning that its evolution is determined
+ * by its initial state, requiring no further input. One interacts with the Game of Life by creating an initial
+ * configuration and observing how it evolves. It is Turing complete and can simulate a universal constructor or any other Turing machine.
+ * @author [Ali Naderi](https://github.com/khod-naderi)
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <unistd.h> // for sleep (usleep)
```
[the remark this answers is not in the document]
Comment (khod-naderi) [reply games/game_of_life.c]: I do not speak English, i explain why I imported these header/library ?

File: games/game_of_life.c [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+ * mathematician John Horton Conway in 1970. It is a zero-player game, meaning that its evolution is determined
+ * by its initial state, requiring no further input. One interacts with the Game of Life by creating an initial
+ * configuration and observing how it evolves. It is Turing complete and can simulate a universal constructor or any other Turing machine.
+ * @author [Ali Naderi](https://github.com/khod-naderi)
+*/
+
+#include <stdio.h> // stundard input/output
+#include <stdlib.h> // for genrate random number AND memory management 
+#include <time.h> // get time system for seed random number
+#include <unistd.h> // for sleep func (usleep)
```
Comment (Panquesito7) [inline games/game_of_life.c:16]:
```suggestion
#include <stdio.h>     /// standard input/output
#include <stdlib.h>    /// to genrate random number AND memory management 
#include <time.h>      /// get time system for seed random number
#include <unistd.h>    /// for sleep func (usleep)
```

File: games/game_of_life.c [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+
+// define Make fixed values
+#define MAX_X 60 // End point X
+#define MAX_Y 20 // End point Y
+
+
+/**
+ * draw program function. (void)
+ * For Show the world
```
Comment (Panquesito7) [inline games/game_of_life.c:25]: Please add proper documentation. See #797 as a reference. Check other parts of the code, too.

File: games/game_of_life.c [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+#include <time.h>      /// get time system for seed random number
+#include <unistd.h>    /// for sleep func (usleep)
+
+// define Make fixed values
+#define MAX_X 60 // End point X
+#define MAX_Y 20 // End point Y
+
+
+/**
+ * draw program function. (void)
```
Comment (Panquesito7) [inline games/game_of_life.c:24]: Please add a proper function description.

File: games/game_of_life.c (hunk 14 above)
Comment (Panquesito7) [reply games/game_of_life.c:25]: Still not fixed.

File: games/game_of_life.c [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+ * For Show the world
+ * Indicates cells with "@"
+ * @param world, pointer to world parameter
```
Comment (Panquesito7) [inline games/game_of_life.c:27]:
```suggestion
 * @param world pointer to world parameter
```

File: games/game_of_life.c [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+    } 
+}
+
+
+/**
+ * em program function.
+ * @returns char pointer new generation on clean exit. 
+ * For next world calculations
+ * It builds a new generation according to the previous generation
+ * @param world, pointer to world parameter
```
Comment (Panquesito7) [inline games/game_of_life.c:54]:
```suggestion
 * @param world pointer to world parameter
```

File: games/game_of_life.c (hunk 2 above)
Comment (Panquesito7) [reply games/game_of_life.c:50]: Still not fixed.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: games/game_of_life.c [hunk 18]
```diff
@@ -0,0 +1,146 @@
...
+ *
+ * @details The Game of Life, also known simply as Life, is a cellular automaton devised by the British
+ * mathematician John Horton Conway in 1970. It is a zero-player game, meaning that its evolution is determined
+ * by its initial state, requiring no further input. One interacts with the Game of Life by creating an initial
+ * configuration and observing how it evolves. It is Turing complete and can simulate a universal constructor or any other Turing machine.
+ * @author [Ali Naderi](https://github.com/khod-naderi)
+*/
+
+#include <stdio.h>     /// standard input/output
+#include <stdlib.h>    /// to genrate random number AND memory management
```
Comment (Panquesito7) [inline games/game_of_life.c:14]:
```suggestion
#include <stdlib.h>    /// to generate random number AND memory management 
```

File: games/game_of_life.c [hunk 19]
```diff
@@ -0,0 +1,146 @@
...
+        }
+    }
+    
+    // return new generation
+    return newWorld;
+}
+
+
+/**
+ * @brief Function main.
```
Comment (Panquesito7) [inline games/game_of_life.c:107]:
```suggestion
 * @brief Main function
```

File: games/game_of_life.c [hunk 20]
```diff
@@ -0,0 +1,146 @@
...
+ * @returns 0 on Exit without error
```
Comment (Panquesito7) [inline games/game_of_life.c:108]:
```suggestion
 * @returns 0 on exit
```

File: games/game_of_life.c [hunk 21]
```diff
@@ -0,0 +1,146 @@
...
+    for(int y=0; y<MAX_Y; y++){
+        for(int x=0; x<MAX_X; x++){
+            *(world + MAX_X*y + x) = ( (rand() <= (RAND_MAX / 10)) ? 1 : 0);
+        }
+    } 
+    
+    
+    /**
+    * The beginning of the eternal cycle of life
+    */
```
Comment (Panquesito7) [inline games/game_of_life.c:138]:
```suggestion
    // The beginning of the eternal cycle of life
```

File: games/game_of_life.c [hunk 22]
```diff
@@ -0,0 +1,146 @@
...
+    while(1){
+        draw(world); // Show the creatures of the world
+        world = em(world); // Calculate the next generation
+        usleep(300000); // A short break = 0.03 sec
```
Comment (Panquesito7) [inline games/game_of_life.c:142]:
```suggestion
        draw(world);           /// Show the creatures of the world
        world = em(world);    /// Calculate the next generation
        usleep(300000);      /// A short break = 0.03 sec
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/khod-naderi/C/actions. **This is something that's required and mentioned in the contributing guidelines.**

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please address **all** the unresolved conversations above (don't only click on `Resolve`, try to really resolve them).

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
By addressing all the unresolved conversations, I didn't mean to only click on `resolve`: I also meant to try to fix them. Try to make your code as per our [guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Thank you.

**Comment by khod-naderi** [comment]
I'm very sorry, my English was not good enough for effective communication

**Pull request closed without merging**